Modules/@babylonjs/react-native/android/include/XrContextHelperARCore.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#if __has_include("jsi/jsi.h")
+#include "jsi/jsi.h"
+#include "IXrContextARCore.h"
+
+namespace BabylonReactNative
+{
+    bool TryGetXrContext(facebook::jsi::Runtime& jsiRuntime, IXrContextARCore*& xrContext)
+    {
+        xrContext = nullptr;
+        if (!jsiRuntime.global().hasProperty(jsiRuntime, "navigator"))
+        {
+            return false;
+        }
+
+        auto navigator{ jsiRuntime.global().getProperty(jsiRuntime, "navigator").asObject(jsiRuntime) };
+        if (!navigator.hasProperty(jsiRuntime, "xr"))
+        {
+            return false;
+        }
+
+        auto nativeXr{ navigator.getProperty(jsiRuntime, "xr").asObject(jsiRuntime) };
+        if (!nativeXr.hasProperty(jsiRuntime, "nativeXrContext") ||
+            !nativeXr.hasProperty(jsiRuntime, "nativeXrContextType") ||
+            nativeXr.getProperty(jsiRuntime, "nativeXrContextType").asString(jsiRuntime).utf8(jsiRuntime) != "ARCore")
+        {
+            return false;
+        }

```

Comment:
Should we move this to a shared header? Is it worth it? Seems like we're starting to get some duplication across platforms and features.